dissect/target/plugins/apps/webservers/apache.py

```diff
@@ -32,43 +128,76 @@ def infer_log_format(line: str) -> Optional[LogFormat]:
         vhost_combined = 'example.com:80 1.2.3.4 - - [19/Dec/2022:17:25:40 +0100] "GET / HTTP/1.1" 200 312 "-"
                           "Mozilla/5.0 (Windows NT 10.0; Win64; x64); AppleWebKit/537.36 (KHTML, like Gecko)
                           Chrome/108.0.0.0 Safari/537.36"'
-    """
 
-    first_part = line.split(" ")[0]
+    Two logformats encountered on Citrix Netscalers:
+        combined_resptime_with_citrix_hdrs  = '1.2.3.4 -> 192.168.4.20 - - [19/Dec/2022:17:25:12 +0100] [12311]
+                                               "GET / HTTP/1.1" 200 712 "-" "Mozilla/5.0
+                                               (Windows NT 10.0; Win64; x64); AppleWebKit/537.36 (KHTML, like Gecko)
+                                               Chrome/108.0.0.0 Safari/537.36" "Time: 11264 microsecs"'
+
+
+        combined_resptime                   = '127.0.0.1 - - [19/Dec/2022:17:25:12 +0100] [69195] "GET / HTTP/1.1" 200
+                                               18705 "-" "curl/7.78.0" "Time: 49835 microsecs"'
+
+    """
+    splitted_line = line.split(" ")
+    first_part = splitted_line[0]
+    second_part = splitted_line[1]
     if ":" in first_part and "." in first_part:
         # does not start with IP, hence it must be a vhost typed log
         return LogFormat.VHOST_COMBINED
+    elif second_part == "->":
+        # Citrix-like
+        return LogFormat.CITRIX_NETSCALER_COMBINED_RESPONSE_TIME_WITH_HEADERS
     elif line[-1:] == '"':
-        # ends with a quotation mark, meaning three is a user agent
+        if "Time: " in line:
+            return LogFormat.CITRIX_NETSCALER_COMBINED_RESPONSE_TIME
+        # ends with a quotation mark but does not contain a response time, meaning there is only a user agent
         return LogFormat.COMBINED
     elif line[-1:].isdigit():
         return LogFormat.COMMON
     return None
 
 
+def apache_response_time_to_ms(time_str: str) -> int:
```

Comment:
You're not always returning an `int` here, due to the division by 1000 case you're dealing with `microsecs`. So either change the return type to `Union[int, float]` or parse the `microsecs` case to an `int`